### PR TITLE
Add documentation for vouch-proxy integration (ref #600)

### PR DESCRIPTION
**Pull request by daedric**
Fixes #

- [ ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ ] cargo test has been run and passes
- [x] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by daedric** [comment]
@Firstyear following our discussion on #600 here is a PR adding some documentation on how to integrate vouch proxy

Let me know if anything is missing
Thanks a lot

File: kanidm_book/src/integrations/oauth2.md [hunk 1]
```diff
@@ -270,4 +270,35 @@ these to a group with a scope map due to Velociraptors high impact.
...
+  client_id: <oauth2_rs_name> # Found in kanidm system oauth2 get XXXX (should be the same as XXXX)
+  client_secret: <oauth2_rs_basic_secret> # Found in kanidm system oauth2 get XXXX
+  code_challenge_method: S256
+  provider: oidc
+  scopes:
+    - login
+    - openid
+    - read_user
+    - profile
+    - email # <<<<< Important
```
Comment (Firstyear) [inline kanidm_book/src/integrations/oauth2.md:295]: Could it be worth explaining why email is important, and what these other scopes are for?

File: kanidm_book/src/integrations/oauth2.md [hunk 2]
```diff
@@ -270,4 +270,35 @@ these to a group with a scope map due to Velociraptors high impact.
...
+  token_url: https://idm.wherekanidmruns.com/oauth2/token
+  user_info_url: https://idm.wherekanidmruns.com/oauth2/openid/lgs/userinfo
```
Comment (Firstyear) [inline kanidm_book/src/integrations/oauth2.md:297]: Is this url correct? I think lgs should be `<oauth2_rs_name>`

**Review by Firstyear — CHANGES_REQUESTED** [review]
Thanks so much, just some minor things, but appreciate your contribution! Make sure to add your self to CONTRIBUTORS.md too :)

File: kanidm_book/src/integrations/oauth2.md (hunk 1 above)
Comment (daedric) [reply kanidm_book/src/integrations/oauth2.md]: Done, the other scopes are not, it was leftovers of me playing around, sorry about that

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
Thanks!

**Pull request merged**